assets/src/components/detours/diversionPage.tsx

```diff
@@ -41,11 +40,27 @@ export const DiversionPage = ({
     routeSegments,
 
     canUndo,
-    undo,
-    clear,
-    finishDetour,
-    editDetour,
-  } = useDetour(originalRoute.routePatternId)
+  } = detourInfo
+
+  const addWaypoint =
+    detourInfo.state === DetourState.Edit ? detourInfo.addWaypoint : undefined
```

Comment:
After chewing on this over the evening, I'm going back and forth on whether I like this approach.

On the one hand, it offends my sensibilities to have a bunch of fields that may or may not be `undefined`.

On the other hand, I'm not sure how much we gain by forcing `diversionPage` (or `DetourMap`) to do this sort of type checking, which creates a lot of syntactic noise and is brittle in its own way, rather than just letting `useDetour` "undefine" the things that it needs to.

I do want to see a move towards things like `addWaypoint` just being defined in the places closest to where it's used, but I'm less convinced than I was earlier that this is the right way to get there.